metashare/api/serializers.py

```diff
@@ -302,16 +362,32 @@ class Meta:
             "latest_commit_at": {"read_only": True},
             "last_checked_unsaved_changes_at": {"read_only": True},
             "url": {"read_only": True},
-            "unsaved_changes": {"read_only": True},
             "currently_refreshing_changes": {"read_only": True},
             "currently_capturing_changes": {"read_only": True},
             "currently_refreshing_org": {"read_only": True},
-            "valid_target_directories": {"read_only": True},
+            "delete_queued_at": {"read_only": True},
+            "owner_gh_username": {"read_only": True},
+            "has_been_visited": {"read_only": True},
         }
 
     def get_has_unsaved_changes(self, obj) -> bool:
         return bool(getattr(obj, "unsaved_changes", {}))
 
+    def get_unsaved_changes(self, obj) -> dict:
+        user = getattr(self.context.get("request"), "user", None)
+        if obj.owner == user:
+            return obj.unsaved_changes
+        return {}
+
+    def get_total_unsaved_changes(self, obj) -> int:
+        return len(obj.unsaved_changes)
```

Comment:
@wlonk Hm, will this work? `unsaved_changes` is an object, where the values are arrays -- I think we want to get the total # of strings in the flattened array of all value arrays.

Reply:
Oh, is that what we really need? I can return that.